src/test/java/codeu/controller/NotificationServletTest.java

```diff
@@ -0,0 +1,108 @@
+  private HttpServletResponse mockResponse;
+  private RequestDispatcher mockRequestDispatcher;
+  private EventStore mockEventStore;
+  private UserStore mockUserStore;
+  private User mockUser;
+
+
+  @Before
+  public void setup() {
+    NotificationServlet = new NotificationServlet();
+
+    mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockSession = Mockito.mock(HttpSession.class);
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+
+    mockResponse = Mockito.mock(HttpServletResponse.class);
+
+    mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/notifications.jsp"))
+        .thenReturn(mockRequestDispatcher);
+
+    mockEventStore = Mockito.mock(EventStore.class);
+    NotificationServlet.setEventStore(mockEventStore);
+
+    mockUserStore = Mockito.mock(UserStore.class);
+    NotificationServlet.setUserStore(mockUserStore);
+
+    mockUser = Mockito.mock(User.class);
+  }
+
+
+  @Test
+  public void tesDoGet() throws IOException, ServletException {
+    List<Event> fakeEventList = new ArrayList<>();
+
+    UUID fakeUserId = UUID.randomUUID();
+    User fakeUser = new User(fakeUserId, "testUser", "test_password_hash", Instant.ofEpochMilli(1000));
+    List<String> userInformation = new ArrayList<>();
+    userInformation.add(fakeUser.getName());
+    Event userEvent = new Event(UUID.randomUUID(), "User", fakeUser.getCreationTime(), userInformation);
+    fakeEventList.add(userEvent);
+
+    UUID fakeConversationId = UUID.randomUUID();
+    Conversation fakeConversation = new Conversation(fakeConversationId, fakeUserId, "testConversation", Instant.ofEpochMilli(2000));
+    List<String> conversationInformation = new ArrayList<>();
+    conversationInformation.add(fakeUser.getName());
+    conversationInformation.add(fakeConversation.getTitle());
+    Event conversationEvent = new Event(UUID.randomUUID(), "Conversation", fakeConversation.getCreationTime(), conversationInformation);
+    fakeEventList.add(conversationEvent);
+
+    Message fakeMessage = new Message(UUID.randomUUID(), fakeConversationId, fakeUserId, "testMessage", Instant.ofEpochMilli(3000));
+    List<String> messageInformation = new ArrayList<>();
+    messageInformation.add(fakeUser.getName());
+    messageInformation.add(fakeConversation.getTitle());
+    messageInformation.add(fakeMessage.getContent());
+    Event messageEvent = new Event(UUID.randomUUID(), "Message", fakeMessage.getCreationTime(), messageInformation);
+    fakeEventList.add(messageEvent);
+
+    Instant fakeLastSeenTime =  Instant.ofEpochMilli(1500);
+    mockUser.setLastSeenNotifications(fakeLastSeenTime);
+
+
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("test_User");
+    Mockito.when(mockUserStore.getUser("test_User")).thenReturn(mockUser);
+    Mockito.when(mockEventStore.getEventsSince(fakeLastSeenTime)).thenReturn(fakeEventList);
```

Comment:
If we don't mock the call to `getEventsSince`, then we should expect the `doGet` call to return just the last two events, since they're both past 1500 ms, right?